src/tarballs/build.ts

```diff
@@ -66,7 +66,21 @@ export async function build(c: BuildConfig, options: {
     const yarnRoot = findYarnWorkspaceRoot(c.root) || c.root
     if (fs.existsSync(path.join(yarnRoot, 'yarn.lock'))) {
       await fs.copy(path.join(yarnRoot, 'yarn.lock'), path.join(c.workspace(), 'yarn.lock'))
-      await exec('yarn --no-progress --production --non-interactive', {cwd: c.workspace()})
+
+      const yarnVersion = (await exec('yarn -v')).stdout.charAt(0)
+      if (yarnVersion === '1') {
+        await exec('yarn --no-progress --production --non-interactive', {cwd: c.workspace()})
+      } else {
+        try {
+          await exec('yarn workspaces focus --production', {cwd: c.workspace()})
+        } catch (error: unknown) {
+          if (error instanceof Error && error.message.includes('Command not found')) {
+            throw new Error('Missing workspace tools. Run `yarn plugin import workspace-tools`.')
+          }
```

Comment:
I wasn't able to get this conditional to correctly identify `Command not found` errors; it seems like the resulting error's message is just `Command failed`, but it has an extra property `stdout` which includes the `"Command not found"` substring.

This is consistent with the [NodeJS docs](https://nodejs.org/api/child_process.html#child_processexeccommand-options-callback) on this usage: 
> In case of an error (including any error resulting in an exit code other than 0), a rejected promise is returned, with the same error object given in the callback, but with two additional properties stdout and stderr.

It's a bit verbose, but this does catch it for me:
```ts
  const isCommandNotFoundError = (error: unknown): boolean => {
    type ErrorWithStdout = Error & {
      stdout: string;
    }
    const typedError = error as ErrorWithStdout
    if (typedError instanceof Error && typeof typedError.stdout === 'string') {
      return typedError.stdout.includes('Command not found')
    }

    return false
  }
```

Reply:
Hi @amccarthy1  - I tried adding a simplified version of your check 
```
  const isCommandNotFoundError = (error: unknown): boolean => {
    type ErrorWithStdout = Error & {
      stdout: string;
    }
    const typedError = error as ErrorWithStdout
    return typedError.stdout.includes('Command not found')
  }
``` 
this worked for yarn1 and yarn3 🎉 

when I was testing with yarn2, and was told to `yarn plugin import workspace-tools` that command failed. I did quite a bit of researching but wasn't able to find a command or version that would install that plugin... if you can figure out a way to install `workspace-tools` in yarn 2 we could add that to the error message... otherwise I think it'd be ok to continue as is.